modules/dlm/src/main/java/org/elasticsearch/dlm/DataLifecycleService.java

```diff
@@ -331,6 +440,79 @@ public void onFailure(Exception e) {
         });
     }
 
+    /*
+     * This method executes the given force merge request. Once the request has completed successfully it writes a timestamp as custom
+     * metadata in the cluster state indicating when the force merge has completed. The listener is notified after the cluster state
+     * update has been made, or when the forcemerge fails or the write of the to the cluster state fails.
+     */
+    private void forceMergeIndex(ForceMergeRequest forceMergeRequest, ActionListener<Void> listener) {
+        assert forceMergeRequest.indices() != null && forceMergeRequest.indices().length == 1 : "DLM force merges one index at a time";
+        final String targetIndex = forceMergeRequest.indices()[0];
+        logger.info("DLM is issuing a request to force merge index [{}]", targetIndex);
+        client.admin().indices().forceMerge(forceMergeRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(ForceMergeResponse forceMergeResponse) {
+                logger.info("DLM successfully force merged index [{}]", targetIndex);
+                setForceMergeCompletedTimestamp(targetIndex, listener);
+            }
```

Comment:
I don't think we can ignore the `forceMergeResponse` here. It includes the number of total, successful, and failed shards, as well as a list of shard failure exceptions. Perhaps this should only be considered successful if `total == successful` for the shard counts in the response?

Reply:
Yeah that's a good point.